tmt/base.py

```diff
@@ -4149,7 +4154,25 @@ def plans(
         ]
 
         if not Plan._opt('shallow'):
-            plans = functools.reduce(operator.iadd, (plan.resolve_imports() for plan in plans), [])
+            unresolved_plans = plans
+            plans = []
+            for plan in unresolved_plans:
+                try:
+                    plans += plan.resolve_imports()
+                except Exception as error:
+                    if self.import_before_filter:
+                        # If we filter later, we can skip some resolve failures
+                        # since it may be unrelated
+                        self.warn(f"Failed to import plan '{plan.name}'")
+                        self.debug("import-fail", str(error))
```

Comment:
Maybe `tmt.utils.show_error_as_warning` instead of a warning and a debug message with the error? It would save the traceback, it would notify user, it would be using the standard report-the-error-but-move-on helper.